### PR TITLE
Add facade docblocks and phpdoc block generation script

### PR DESCRIPTION
**Pull request by srtfisher**
Adds a Facade docblock generation script from Laravel that can ensure that all Facades have `@method` on them. Also adds docblocks to make IDE experience a bit better. These docblocks on the facades and model should make it better to do something like `Post::whereId(1234)->first()` and have your IDE know that your being returned `Post` vs a generic model instance (no pun intended).

Feel free to ignore `bin/docblock.php` from review, it's for internal use.

### CODE REVIEW

File: src/mantle/facade/class-request.php [hunk 1]
```diff
@@ -10,6 +10,143 @@
...
+ * @method static void setLocale(string $locale)
+ * @method static string getLocale()
+ * @method static bool isMethod(string $method)
+ * @method static bool isMethodSafe()
+ * @method static bool isMethodIdempotent()
+ * @method static bool isMethodCacheable()
+ * @method static string|null getProtocolVersion()
+ * @method static string|resource getContent(bool $asResource = false)
+ * @method static \Symfony\Component\HttpFoundation\InputBag getPayload()
+ * @method static array toArray()
```
Comment (mogmarsh) [inline src/mantle/facade/class-request.php:105]: is this different from to_array on line 40?

**Review by mogmarsh — APPROVED** [review]
wow, this was a lot to review. I found a few small things, but overall this looks good. 👍

File: src/mantle/facade/class-request.php (hunk 1 above)
Comment (srtfisher) [reply src/mantle/facade/class-request.php:105]: yes, it has both

File: src/mantle/facade/class-app.php [hunk 2]
```diff
@@ -9,6 +9,86 @@
...
+ * @method static mixed build(\Closure|string $concrete)
+ * @method static void resolving(\Closure|string $abstract, \Closure|null $callback = null)
+ * @method static void after_resolving(\Closure|string $abstract, \Closure|null $callback = null)
+ * @method static array get_bindings()
+ * @method static string get_alias(string $abstract)
+ * @method static void forget_extenders(string $abstract)
+ * @method static void forget_instance(string $abstract)
+ * @method static void forget_instances()
+ * @method static \Mantle\Contracts\Container getInstance()
+ * @method static \Mantle\Contracts\Container get_instance()
```
[the remark this answers is not in the document]
Comment (srtfisher) [reply src/mantle/facade/class-app.php:81]: Yep!

File: src/mantle/facade/class-app.php [hunk 3]
```diff
@@ -9,6 +9,86 @@
...
+ * @method static void flush()
+ * @method static void bootstrap_with(array $bootstrappers, \Mantle\Contracts\Kernel $kernel)
+ * @method static bool is_booted()
+ * @method static \Mantle\Application\Application boot()
+ * @method static string environment_file(string $file = null)
+ * @method static string environment_path(string $path = null)
+ * @method static string environment()
+ * @method static bool is_environment(string|array ...$environments)
+ * @method static string get_namespace()
+ * @method static string namespace()
```
[the remark this answers is not in the document]
Comment (srtfisher) [reply src/mantle/facade/class-app.php:40]: no difference, same thing. the code generated all of this

**Pull request merged**